libraries/netapp_e_series_api.rb

```diff
@@ -4,11 +4,12 @@
 class NetApp
   class ESeries
     class Api
-      def initialize(user, password, url, basic_auth, connect_timeout = nil)
+      def initialize(user, password, url, basic_auth, asup = true, connect_timeout = nil)
```

Comment:
The calls in #netapp_api_create in libraries/helper.rb that create new instances of NetApp::ESeries::Api do not make use of the node['netapp']['asup'] attribute.  This causes two problems:

 * the user cannot disable the ASUP call because you're defaulting to true in your #initialize method, and
 * the user cannot set a timeout because the call in helper.rb is putting the timeout in place of the asup variable since the asup variable was added *before* connect_timeout.

Before we merge this, we should make the following changes:

 * The asup variable here in #initialize should have no default.
 * The #netapp_api_create method in libraries/helper.rb should be modified to pass in the asup node attribute
 * The tests should be modified accordingly, accounting for both the new parameter for #initialize and the fact that the timeout parameter has been moved.

An alternative path to adding the asup variable to #initialize would be to change the input parameters to be a hash, and then the hash is processed in #initialize where defaults can be enforced for more than just one input parameter.  However, we still need to modify #netapp_api_create accordingly.

Thanks,
~Adam